README.md

```diff
@@ -142,14 +146,24 @@ Model.reset_column_information
 
 # JavaScript
 * Use CoffeeScript over JavaScript
+* Namespace javascript objects for related chuncks of funcationlity (Components, Engines, etc...)  [example implementation](samples/js_namespace.md)
+* Implement an initializers file applied to any content loaded on the page [see example](samples/initializers.js.md)
+* Use js classes and encapsulation wherever possible ([class example](samples/js_class.md), [encapsulation example](samples/js_encapsulation.md))
 * Use a package manager like Bower for installing dependencies? (This item is questionable as some people raised issues with using Bower with Rails. We can discuss further.)
 * Prefer a JavaScript framework over vanilla JavaScript. (This means don't roll your own custom event library or other things that exist out there. But do make sure the company is on board with any new libraries that are used before including them in a project)
 * Use IIFE or Object Literal notation
 * Separate responsibilities of vanilla JavaScript into separate entities.
 * Query DOM elements using `js-` or `data-` attributes instead of CSS classes
 * Do not rely on or store data in the DOM. (This is particularly true from a security stand point. Do not put secure or sensitive information in the DOM. For example do not have a hidden field with a SSN, or that determines whether someone is an admin).
+* Coffeescript file extensions not include `.js` where possible. Prefer `foo.coffee` to `foo.js.coffee.`
```

Comment:
As long as it transpiles appropriately, I would say we should investigate ES6 and use native JS classes.  

It is worth noting that Chrome does not support ES6 classes at the moment.